### PR TITLE
Adding new data models

### PR DESCRIPTION
**Pull request by IIsmahan**
Added the BlogLike, BlogComment and user models in.

### CODE REVIEW

File: models/index.js
Comment (Luke-G91) [inline models/index.js:1]: this can be removed

File: models/BlogPost.js [hunk 1]
```diff
@@ -13,15 +13,20 @@ BlogPost.init(
       type: DataTypes.TEXT,
       allowNull: false,
     },
-    author: {
-      type: DataTypes.STRING(100),
+    user_id: {
```
Comment (Luke-G91) [inline models/BlogPost.js]: ```userId``` instead it was wrong on the ERD

File: models/BlogUser.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+class BlogUser extends Model {}
+
+BlogComment.init(
+  {
+    ID: {
+      type: DataTypes.INTEGER,
+      primaryKey: true,
+      autoIncrement: true,
+    },
```
Comment (Luke-G91) [inline models/BlogUser.js]: pkey is auto generated, so we shouldn't need this atm

File: models/BlogUser.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      allowNull: true, 
+    },
+    hashedPassword: {
+      type: DataTypes.STRING(255),
+      allowNull: false,
+    },
+    profilePicture: {
+      type: DataTypes.TEXT, 
+      allowNull: true,
+    },
```
Comment (Luke-G91) [inline models/BlogUser.js]: we won't need this unless we decide to add the profile pictures, maybe remove for now?

File: models/BlogUser.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+const { DataTypes, Model } = require("sequelize");
+const sequelize = require("../config/database");
+
+class BlogUser extends Model {}
+
+BlogComment.init(
```
Comment (Luke-G91) [inline models/BlogUser.js]: Should be User.init

File: models/BlogUser.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+const { DataTypes, Model } = require("sequelize");
+const sequelize = require("../config/database");
+
+class BlogUser extends Model {}
```
Comment (Luke-G91) [inline models/BlogUser.js]: Can just be User

File: models/BlogLike.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+class BlogLike extends Model {}
+
+BlogLike.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      autoIncrement: true,
+      primaryKey: true,
+    },
```
Comment (Luke-G91) [inline models/BlogLike.js]: this should be auto generated so can be removed for now

File: models/BlogLike.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    blog_id: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+    user_id: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+  },
```
Comment (Luke-G91) [inline models/BlogLike.js]: ```blogId``` and ```userId``` instead it was wrong on the ERD

File: models/BlogLike.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  {
+    sequelize,
+    modelName: "BlogLike",
+    tableName: "BlogLikes",
```
Comment (Luke-G91) [inline models/BlogLike.js]: table name can be kept non plural

File: models/BlogComment.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+BlogComment.init(
+  {
+    blog_id: {
+      type: DataTypes.INTEGER(),
+      allowNull: false,
+    },
+    user_id: {
+      type: DataTypes.INTEGER(),
+      allowNull: false,
+    },
```
Comment (Luke-G91) [inline models/BlogComment.js]: ```blogId``` and ```userId``` instead it was wrong on the ERD

**Review by Luke-G91 — CHANGES_REQUESTED** [review]

File: models/BlogUser.js (hunk 3 above)
Comment (Ben-cyber-hub) [reply models/BlogUser.js]: can be kept, is allowed to be null and ignored for now

**Review by Luke-G91 — APPROVED** [review]

**Pull request merged**